test/util_test.sh

```diff
@@ -0,0 +1,300 @@
+#!/bin/bash
+# This file ensures that the helper functions in util.sh behave as expected
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+OS_ROOT=$(dirname "${BASH_SOURCE}")/..
+source "${OS_ROOT}/hack/util.sh"
+source "${OS_ROOT}/test/util.sh"
+os::log::install_errexit
+
+function echo_and_exit() {
+	echo $1
+	exit $2
+}
+
+mkdir -p /tmp/openshift/origin/test/cmd/
+JUNIT_OUTPUT_FILE=/tmp/openshift-cmd/junit_output.txt
+
+VERBOSE=1
+
+# positive tests
+os::test::expect_success "exit 0"
+
+os::test::expect_failure "exit 10"
+
+os::test::expect_success_and_text "echo_and_exit hello 0" "hello"
```

Comment:
I tried for a very long time to get the second style to work, and I could not. It would always echo everything after that command. I'd appreciate help if you feel strongly about it. @Miciah didn't have a suggestion here.